Source/BugsnagCrashReport.m

```diff
@@ -579,29 +580,38 @@ - (NSArray *)serializeThreadsWithException:(NSMutableDictionary *)exception {
                         BSGFormatFrame(mutableFrame, [self binaryImages]));
                 }
             }
-
             BSGDictSetSafeObject(exception, stacktrace, BSGKeyStacktrace);
-        } else {
-            NSMutableArray *threadStack = [NSMutableArray array];
+        }
+        [self serialiseThread:bugsnagThreads thread:thread backtrace:backtrace crashedThread:isCrashedThread];
+    }
+    return bugsnagThreads;
+}
 
-            for (NSDictionary *frame in backtrace) {
+- (void)serialiseThread:(NSMutableArray *)bugsnagThreads
+                 thread:(NSDictionary *)thread
+              backtrace:(NSArray *)backtrace
+          crashedThread:(BOOL)isCrashedThread {
```

Comment:
I would be careful to call this the reportingThread or the currentThread to emphasize that its not the crashing thread in all cases.